convert/convert.go

```diff
@@ -611,3 +617,18 @@ func (b *builder) handleLiteralString(lit proto.Literal) (string, error) {
 	}
 	return lit.Source, nil
 }
+
+func (b *builder) validatePackageName() error {
+	pkgName := b.pkg.Name
+	for _, c := range pkgName {
+		switch {
+		case c >= '0' && c <= '9':
+		case c >= 'a' && c <= 'z':
+		case c >= 'A' && c <= 'Z':
+		case c == '_':
+		default:
+			return fmt.Errorf("invalid character %q in package name %q", c, pkgName)
+		}
+	}
+	return nil
```

Comment:
this is slightly incorrect :) package names are just identifiers: https://golang.org/ref/spec#Package_clause

So, for example, `"123"` and `""` are invalid, but `"àé"` is valid. I realise that package names in theory shouldn't be empty or just numbers to begin with, but this still makes me uneasy.

Unlucky for us, there's no API in `go/ast` or `astutil` to easily check this. For now, I'd just add this func to the loader package:

```
func IsIdentifier(str string) bool {
	if len(str) == 0 {
		return false
	}
	for i, c := range str {
		if unicode.IsLetter(c) || c == '_' || (i > 0 && unicode.IsDigit(c)) {
			continue
		}
		return false
	}
	return true
}
```

Reply:
> I  realise that package names in theory shouldn't be empty or just numbers to begin with, but this still makes me uneasy.

The proto language doesn't allow empty package names, or package names starting with a number, so the proto parser we are using will error on those scenarios (which we have been relying on previously as well).

I wanted to keep the position information of the error just to be useful, and since this won't be used anywhere else yet I will keep this function in this package; if / when we need to use this somewhere else we can look at moving this to the loader or wherever seems appropriate.

Reply:
Actually, ignore me, I removed the positional information of the error and didn't realise it. I will add this to the loader.

Reply:
I ended up keeping a slightly modified version of your code in the convert package. I decided to keep the character that was causing issues just incase people don't know exactly what a Go identifier is.

Reply:
SGTM! But did you forget to use `loader.IsIdentifier`? I think it should be fine to quote the entire string in the error, like:

> invalid package name %q: identifiers may only contain letters, digits, and underscores

Reply:
Ah nope, I decided to leave the identifier check in this function (just to keep information about which character was incorrect).

Reply:
Ok, but then I don't see the point of adding the other func :) Merging for now, anyway.

Reply:
I don't know wha other func yo are referring to here, this one is the only one I added :)

Reply:
https://github.com/gunk/gunk/pull/199/files#diff-b38e43f0fb016af3e8cecd2d12d779ecR705 :)

Reply:
Oh, that is most definitely not supposed to be there, my apologies :(

Reply:
No worries! Want to send a follow-up PR?